woocommerce/woocommerce-android
# Issue/8282 remove xmlrpc checks

Closes: #8282 

### Description
This PR updates the REST API login to use Cookie Nonce authentication instead of doing the XMLRPC discovery and authentication first.
It builds on top of the following PRs:
1. https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2643
2. https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2649
3. https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/109

The FluxC changes were already merged and are part of the app's current version.

### Testing instructions
All of the tests below require the `treatment` variant of the REST API experiment, so please make sure you change the returned variant [here](https://github.com/woocommerce/woocommerce-android/blob/c3bd5d6cb2ccc54f51201745ca28932cdfa4e016/WooCommerce/src/main/kotlin/com/woocommerce/android/experiment/RESTAPILoginExperiment.kt#L47) accordingly.
##### Happy path
1. Open the app and click on "Enter site address".
2. Enter the site address.
3. Enter the site credentials.
5. Confirm the login succeeds.

##### Wrong credentials
1. Open the app and click on "Enter site address".
2. Enter the site address.
3. Enter some wrong site credentials.
4. Confirm the correct error message is shown.

##### WooCommerce not installed
1. Disable or uninstall WooCommerce from your site.
2. Open the app and click on "Enter site address".
3. Enter the site address.
4. Enter the site credentials.
6. Confirm the WooCommerce Not Installed error screen is shown.

##### Application Passwords disabled
1. Install this [plugin ](https://wordpress.org/plugins/disable-application-passwords/) to disable Application Passwords.
2. Open the app and click on "Enter site address".
3. Enter the site address.
4. Enter the site credentials.
5. Confirm the Application Passwords error screen is shown.

##### User role not eligible
1. Update or create a new user with a role other than `admin` and `show-manager`.
2. Open the app and click on "Enter site address".
3. Enter the site address.
4. Enter the site credentials.
5. Confirm the user role error is shown.

##### XMLRPC disabled
1. Install this [plugin](https://wordpress.org/plugins/manage-xml-rpc/), then check "Disable XML-RPC" in XML-RPC settings page.
2. Open the app and click on "Enter site address".
3. Enter the site address.
4. Enter the site credentials.
5. Confirm the login succeeds.
